File: lib/rest/yampi/mappers/yampi_order_item_mapper.dart
import 'package:sertton/core/checkout/dtos/order_item_dto.dart';
import 'package:sertton/rest/types/json.dart';

class YampiOrderItemMapper {
  static OrderItemDto toDto(Json json) {
    final skuData = json['sku']?['data'] ?? {};

    return OrderItemDto(
      id: json['id'].toString(),
      quantity: json['quantity'] ?? 0,
      price: double.tryParse(json['price'].toString()) ?? 0.0,
      skuName: skuData['title'] ?? json['sku_name'] ?? '',
      skuCode: skuData['sku'] ?? json['sku_code'] ?? json['item_sku'] ?? '',
      skuSalePrice:
          double.tryParse(
            (skuData['price_sale'] ?? json['price_sale']).toString(),
          ) ??
          0.0,
      skuDiscountPrice:
          double.tryParse(
            (skuData['price_discount'] ?? json['price_discount']).toString(),
          ) ??
          0.0,
    );
  }

  static List<OrderItemDto> toDtoList(Json json) {
    if (json.containsKey('data')) {
      final data = (json['data'] as List).cast<Json>();
      return data.map(toDto).toList();
    }
    return [];
  }
}

File: sertton-react-native-app/src/rest/yampi/mappers/yampi-order-item-mapper.ts
import type { OrderItem } from "@/core/checkout/entities"
import type { YampiOrderItem } from "@/rest/yampi/types"

export const YampiOrderItemMapper = () => ({
  toDomain(input: YampiOrderItem): OrderItem {
    const sku = input.sku && "data" in input.sku ? input.sku.data : input.sku

    return {
      id: String(input.id),
      quantity: input.quantity,
      price: input.price,
      skuName: sku?.title ?? input.sku_name ?? input.item_sku,
      skuCode: sku?.sku ?? input.sku_code ?? input.item_sku,
      skuSalePrice: sku?.price_sale ?? input.price_sale ?? input.price,
      skuDiscountPrice: sku?.price_discount ?? input.price_discount ?? input.price,
    }
  },
})
